### ISSUE TITLE
Add a rule to validate servers in the channel

### ISSUE THREAD

**Issue by magicmatatjahu** [stoplightio/spectral#2124]
**User story.**
As a user I want to know if the servers defined in the `channels.*.servers` array exist in the `servers` object.

**Is your feature request related to a problem?**
The following AsyncAPI document is invalid and Spectral should throw errors:

```yaml
asyncapi: "2.0.0"
info:
  title: Awesome API
  description: A very well defined API
  version: "1.0"
servers:
  production:
    url: "stoplight.io"
    protocol: "https"
channels:
  hello:
    servers:
      - development
```

**Additional context**
Part of #2100
